internal/db/db.go

```diff
@@ -126,12 +126,12 @@ func (db *DB) Transaction(ctx context.Context, f func(context.Context, *sql.Tx)
 	})
 }
 
-func (db *DB) retry(f func() error) error {
+func (db *DB) retry(f func(context.Context) error) error {
 	if db.ctx.Err() != nil {
-		return f()
+		return f(context.TODO())
 	}
 
-	return query.Retry(f)
+	return query.Retry(context.TODO(), f)
```

Comment:
Now that these methods accept a context, we should pass the given context into the function wherever possible.